lib/elixir/lib/protocol.ex

```diff
@@ -665,10 +665,13 @@ defmodule Protocol do
           end
       end
 
+    fun_arities = :sets.from_list(protocol.__protocol__(:functions), version: 2)
```

Comment:
The set might be overkill since most protocols should define very few functions, so perhaps just a list is fine?

Reply:
Unfortunately we cannot load the code, that would add a bunch of unnecessary work, given we would have to purge it immediately after consolidation anyway. What I proposed is to find the `{:__protocol__, 1}` entry inside `definitions`, then find the `:functions` clauses in there, and get its return type, which should be a keyword list. :)

Reply:
Got it! [6ff5011](https://github.com/elixir-lang/elixir/pull/14191/commits/6ff5011893d42ec20789607f6dadbd426b34fa16)